### PR TITLE
Update nayduck test to reduce transaction failure rate

### PR DESCRIPTION
**Pull request by tayfunelmas**
In the current version of the `memtrie_disktrie_switch.py`, we create the accounts using the same validator's signing key. Since the create-account transactions are issued in parallel, they may be reordered and the chain can see non-increasing sequence of nonces. This is seen when running the test multiple times, where it fails when creating the accounts first.

To avoid this, we use different validators' signing keys to create the accounts instead of using the same account, so different nonces are used.

### CODE REVIEW

File: pytest/tests/sanity/memtrie_disktrie_switch.py [hunk 1]
```diff
@@ -244,38 +251,39 @@ def __deploy_contracts(self):
...
         Waits for the create-account transactions to complete."""
+
+        # Update nonces for the validator accounts, as we will use them to create the new accounts.
+        self.__retrieve_nonces([node.signer_key for node in self.nodes])
+
+        create_account_tx_list = []
         account_keys = []
-        for account_id in ALL_ACCOUNTS:
+        for i in range(len(ALL_ACCOUNTS_PREFIXES)):
+            # Choose a validator node to sign the transaction for creating the new account.
```
Comment (saketh-are) [inline pytest/tests/sanity/memtrie_disktrie_switch.py:263]: One of the nodes in this test is a non-validator

File: pytest/tests/sanity/memtrie_disktrie_switch.py [hunk 2]
```diff
@@ -244,38 +251,39 @@ def __deploy_contracts(self):
...
+            # We do not use a single validator, since transactions and their nonces can be
+            # reordered, which will make the transactions with smaller nonces invalid.
+            signer_key = self.nodes[i % len(self.nodes)].signer_key
```
Comment (saketh-are) [inline pytest/tests/sanity/memtrie_disktrie_switch.py:266]: If this code is used only for this test, in which we have 5 account and 5 nodes, can we just write `i` here?

File: pytest/tests/sanity/memtrie_disktrie_switch.py [hunk 3]
```diff
@@ -135,13 +135,20 @@ def __next_target_height(self, num_epochs):
...
         assert signer_key in self.nonces
         nonce = self.nonces[signer_key]
         self.nonces[signer_key] = nonce + 42
         return nonce
 
+    def __retrieve_nonces(self, account_keys):
+        """Retrieves the next nonce for the accounts and stores them locally."""
+        for account_key in account_keys:
+            nonce = self.rpc_node.get_nonce_for_pk(account_key.account_id,
+                                                   account_key.pk) + 42
```
Comment (saketh-are) [inline pytest/tests/sanity/memtrie_disktrie_switch.py:149]: Why do we need to add 42?

**Review by saketh-are — APPROVED** [review]

File: pytest/tests/sanity/memtrie_disktrie_switch.py (hunk 1 above)
Comment (tayfunelmas) [reply pytest/tests/sanity/memtrie_disktrie_switch.py:263]: updated comment.

File: pytest/tests/sanity/memtrie_disktrie_switch.py (hunk 2 above)
Comment (tayfunelmas) [reply pytest/tests/sanity/memtrie_disktrie_switch.py:266]: yeah i should be sufficient. done.

File: pytest/tests/sanity/memtrie_disktrie_switch.py (hunk 3 above)
Comment (tayfunelmas) [reply pytest/tests/sanity/memtrie_disktrie_switch.py]: adding 1 should work, I just wanted to add a more arbitrary number. checked with +1 only and it worked. done.

**Pull request merged**